src/main/java/org/springframework/samples/petclinic/owner/PetController.java

```diff
@@ -145,11 +146,13 @@ public String processUpdateForm(Owner owner, @Valid Pet pet, BindingResult resul
 		if (result.hasErrors()) {
 			return VIEWS_PETS_CREATE_OR_UPDATE_FORM;
 		}
-
-		owner.addPet(pet);
-		this.owners.save(owner);
-		redirectAttributes.addFlashAttribute("message", "Pet details has been edited");
-		return "redirect:/owners/{ownerId}";
+		if (result.hasErrors()) {
+			return VIEWS_PETS_CREATE_OR_UPDATE_FORM;
+		} else {
+			owner.addPet(pet);
+			this.owners.save(owner);
+			redirectAttributes.addFlashAttribute("message", "Pet details has been edited");
+			return "redirect:/owners/{ownerId}";
+		}
```

Comment:
This change (and all the other changes in this file) are unnecessary; note it introduced code duplication (the `result.hasErrors()` is checked twice), and the else branch was already an implicit `else` due to the return in the check.